infra/auth0/js/setCustomIdTokenClaims.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Handler that will be called during the execution of a PostLogin flow.
+ *
+ * @param {Event} event - Details about the user and the context in which they are logging in.
+ * @param {PostLoginAPI} api - Interface whose methods can be used to change the behavior of the login.
+ */
+exports.onExecutePostLogin = async (event, api) => {
+  const namespace = "https://online.ntnu.no"
+  const { gender, phone, allergies } = event.user.app_metadata
+  const { study_year } = event.user.user_metadata
```

Comment:
Doesn't this crash if study_year is not present?

Reply:
I thought about it but IMO it's okay to just have it as undefined if it is not set?

Reply:
If the attributes don't exist the values will resolve to undefined, which again will resolve to null in the JWT I assume. I'll test it. I think it makes a lot of sense that the values are null

Reply:
This code will raise an exception if app_metadata or user_medatata is undefined, but this should never be the case